### ISSUE TITLE
Lost focus after running mission control or moving space

### ISSUE THREAD

**Issue by songsanli** [railwaycat/homebrew-emacsmacport#124]
There some strange behaviors when use emacs-mac with Mission Control:

### Case 1

Steps
1. I'm using Emacs app
2. I trigger Mission Control to try to do navigation
3. I key the <kbd>esc</kbd> because I give up doing navigation and want to move back to Emacs

Expected Behavior: Mission Control exits and current used application is Emacs.

Actual Behavior: Mission Control exits but current used application will change to recently used application instead of Emacs (as if I clicked the <kbd>cmd</kbd> + <kbd>tab</kbd>).

### Case 2

Steps
1. I'm using Emacs app in Desktop 1
2. I move my desktop to Desktop 2
3. I move back to Desktop 1

Expected Behavoir: I'm now at Desktop 1 and current used application is Emacs.

Actual Behavior: I'm now at Desktop 1 but current used application has been changed to recently used application instead of Emacs (as if I clicked the <kbd>cmd</kbd> + <kbd>tab</kbd>).

------

Is there wrong config in my .emacs.d or a bug in emacs-port?

**Comment by vutran1710** [comment]
Nobody care to answer this?

**Comment by dotemacs** [comment]
Is the behaviour the same with regular Emacs?

If it is, then it's a good chance that is down to the window manager rather than something to do with Emacs.

**Comment by vutran1710** [comment]
nope, regular Emacs works just fine. Plus, its window can maximize like any other apps, but this Emacs cant either.

**Comment by ackerleytng** [comment]
Have this issue too!

**Comment by neezer** [comment]
Same issue here, very annoying:

![2018-06-25 12 20 50](https://user-images.githubusercontent.com/29997/41870980-710dfcaa-7872-11e8-9165-92a646c9487c.gif)

I used the exact same repro steps @songsanli outlined above. 

- railwaycat/emacsmacport/emacs-mac: stable emacs-26.1-z-mac-7.1, HEAD
- macOS 10.13.4

FWIW, I can <kbd>⌘-Tab</kbd> back to the window, but if I select Emacs with the mouse from Mission Control, it immediately backgrounds itself.

**Comment by ackerleytng** [comment]
Oh I found a fix somewhere else on the internet a few days ago! It's because I have menu bar set to off in my config, and osx's menu bar is in the system menu bar anyway, so something weird happens. Turning menu bar back on fixes it for me.

**Comment by neezer** [comment]
@ackerleytng Ahh, that works for me as well:

```diff
- (menu-bar-mode -1)
```

Still kinda feels like a bug worth fixing, though. My two cents.

**Comment by simonmichael** [comment]
Good clue @ackerleytng ! (menu-bar-mode 1) also fixes #139.

**Comment by sa1986meister** [comment]
I still had the same problem a few days ago, but below just fixed this:
`killall Dock`
I don't know if this is due to Emacs-mac since sometimes the same thing happens with other apps.
